docs/latest/design/bootstrap.md

```diff
@@ -0,0 +1,374 @@
+# Bootstrap Design
+
+## Overview
+
+[Issue 31][] specifies the need for allowing advanced users to specify their custom
+Envoy Bootstrap configuration rather than using the default Bootstrap configuration
+defined in Envoy Gateway. This allows advanced users to extend Envoy Gateway and
+support their custom use cases such setting up tracing and stats configuration
+that is not supported by Envoy Gateway.
+
+## Goals
+
+* Define an API field to allow a user to specify a custom Bootstrap
+* Provide tooling to allow the user to generate the default Bootstrap configuration
+  as well as validate their custom Bootstrap.
+
+## Non Goals
+
+* Allow user to configure only a section of the Bootstrap
+
+## API
+
+Leverage the existing [EnvoyProxy][] resource which can be attached to the [GatewayClass][] using
+the [parametersRef][] field, and define a `Bootstrap` field within the resource. If this field is set,
+the value is used as the Bootstrap configuration for all managed Envoy Proxies created by Envoy Gateway.
+
+```go
+// EnvoyProxySpec defines the desired state of EnvoyProxy.
+type EnvoyProxySpec struct {
+    ......
+	// Bootstrap defines the Envoy Bootstrap as a YAML string.
+	// Visit https://www.envoyproxy.io/docs/envoy/latest/api-v3/config/bootstrap/v3/bootstrap.proto#envoy-v3-api-msg-config-bootstrap-v3-bootstrap
+	// to learn more about the syntax.
+	// Some fields within the Bootstrap that are required to communicate with the xDS Server (Envoy Gateway) and receive xDS resources
+	// from it are not configurable and will result in the `EnvoyProxy` resource being rejected.
+	// We strongly recommend using `egctl x translate` to generate a `EnvoyProxy` resource with the `Bootstrap` field set to the default
+	// Bootstrap configuration used. You can edit this configuration, and rerun `egctl x translate` to ensure there are no validation errors.
+	Bootstrap string `json:"bootstrap,omitempty"`
+}
+```
+
+## Tooling
+
+A CLI tool `egctl x translate` will be provided to the user to help generate a working Bootstrap configuration.
+Here is an example where a user inputs a `GatewayClass` and the CLI generates the `EnvoyProxy` resource with the `Bootstrap` field populated.
+
+```
+cat <<EOF | egctl x translate --from gateway-api --to gateway-api -f -
```

Comment:
prefer to `egctl x translate bootstrap`

Reply:
In https://github.com/envoyproxy/gateway/pull/1066, we use `egctl x translate --from gateway-api --to gateway-api -t bootstrap -f xxx` to generate boostrap config of input.

Also we can use:
+ `egctl x translate --from gateway-api --to gateway-api -t cluster`
+ `egctl x translate --from gateway-api --to gateway-api -t listener`
+ `egctl x translate --from gateway-api --to gateway-api -t route`

To get other resource type configs.

Reply:
@Xunzhuo  imho `-t bootstrap` doesnt make sense when output type is `gateway-api`

Reply:
@zirain I would like to continue to leverage `egctl x translate --from gateway-api --to gateway-api` so that in the future we can use this for #24 where we can translate Gateway API resources into Gateway API resources with xDS Filter Extensions

Reply:
@arkodg Right, only works when --to=xds for now.